### PR TITLE
feat: export parser interfaces in Interfaces

### PR DESCRIPTION
**Pull request by RasPhilCo**

### CODE REVIEW

File: src/interfaces/parser.ts [hunk 1]
```diff
@@ -0,0 +1,171 @@
+import {AlphabetLowercase, AlphabetUppercase} from './alphabet'
+import {Config} from './config'
+
+export type ParseFn<T> = (input: string) => T
+
+// eslint-disable-next-line @typescript-eslint/interface-name-prefix
```
Comment (amphro) [inline src/interfaces/parser.ts:6]: This eslint diable is no longer needed right?

File: src/interfaces/parser.ts [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+  required: false;
+  value?: T;
+}
+
+export type ParserArg<T> = RequiredArg<T> | OptionalArg<T>
+
+export interface FlagOutput { [name: string]: any }
+export type ArgInput = Arg<any>[]
+
+// eslint-disable-next-line @typescript-eslint/interface-name-prefix
```
Comment (amphro) [inline src/interfaces/parser.ts:42]: This is no longer needed?

**Review by amphro — APPROVED** [review]

File: src/interfaces/parser.ts (hunk 1 above)
Comment (RasPhilCo) [inline src/interfaces/parser.ts:6]:
```suggestion
```

File: src/interfaces/parser.ts (hunk 2 above)
Comment (RasPhilCo) [reply src/interfaces/parser.ts:41]:
```suggestion
```

**Pull request merged**